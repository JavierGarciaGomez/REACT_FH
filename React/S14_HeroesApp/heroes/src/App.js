// 171

import { AppRouter } from "./routers/AppRouter";

function App() {
  return (
    <div>
      <h1>Heroes App</h1>
      <AppRouter />
    </div>
  );
}

export default App;
