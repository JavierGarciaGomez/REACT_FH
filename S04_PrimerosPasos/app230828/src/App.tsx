import { useState } from "react";

type Props = { value?: number };

const App = ({ value = 100 }: Props) => {
  const handleAdd = (): void => {
    setcount((value) => value + 1);
  };

  const handleSubtract = (): void => {
    setcount((value) => value - 1);
  };

  const handleReset = (): void => {
    setcount(value);
  };

  const [count, setcount] = useState<number>(value);

  return (
    <>
      <h1>{count}</h1>
      <div>DIV</div>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

export default App;
