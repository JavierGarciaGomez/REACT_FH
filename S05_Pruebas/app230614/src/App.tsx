import { MouseEvent, useState } from "react";

type Props = { value: number };

export const App = ({ value }: Props) => {
  const handleAdd = (): void => {
    setcount((value) => value + 1);
  };

  const handleSubtract = (): void => {
    setcount((value) => value - 1);
  };

  const handleReset = (): void => {
    setcount(value);
  };

  const [count, setcount] = useState<number>(value);

  return (
    <>
      <h1>{count}</h1>
      <div>DIV</div>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};
