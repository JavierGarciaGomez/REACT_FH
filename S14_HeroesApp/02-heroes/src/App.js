// 171, 188

import { useEffect } from "react";
import { AppRouter } from "./routers/AppRouter";

function App() {
  return <AppRouter />;
}

export default App;
