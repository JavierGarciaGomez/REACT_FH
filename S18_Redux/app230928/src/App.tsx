import { useDispatch, useSelector } from "react-redux";
import { RootState } from "./store";
import {
  decrement,
  increment,
  incrementByAmount,
} from "./store/slices/counter";

export const App = () => {
  const count = useSelector((state: RootState) => state.counter.value);
  const dispatch = useDispatch();
  return (
    <div>
      <div>
        <span>{count}</span>
        <button
          aria-label="Increment value"
          onClick={() => dispatch(increment())}
        >
          Increment
        </button>
        <button
          aria-label="Increment value by 2"
          onClick={() => dispatch(incrementByAmount(2))}
        >
          Increment by 2
        </button>

        <button
          aria-label="Decrement value"
          onClick={() => dispatch(decrement())}
        >
          Decrement
        </button>
      </div>
    </div>
  );
};
