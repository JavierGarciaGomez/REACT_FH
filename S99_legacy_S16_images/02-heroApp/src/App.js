// 171, 188

import { useEffect, useReducer } from "react";
import { AuthContext } from "./auth/AuthContext";
import { authReducer } from "./auth/authReducer";
import { AppRouter } from "./routers/AppRouter";

const init = () => {
  console.log(localStorage.getItem("user"));
  return JSON.parse(localStorage.getItem("user")) || { logged: false };
};

function App() {
  const [userState, dispatch] = useReducer(authReducer, {}, init);

  useEffect(() => {
    if (!userState) return;

    localStorage.setItem("user", JSON.stringify(userState));
  }, [userState]);

  return (
    <AuthContext.Provider value={{ userState, dispatch }}>
      <AppRouter />
    </AuthContext.Provider>
  );
}

export default App;
