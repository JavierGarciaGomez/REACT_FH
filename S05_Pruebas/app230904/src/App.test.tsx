import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";

describe("pruebas en <App />", () => {
  it("should match the snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("should have a starting value of 100", () => {
    render(<App />);

    expect(screen.getByText("100")).toBeTruthy();
  });

  test("debe de incrementar con el botón +1", () => {
    render(<App value={10} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText("11")).toBeTruthy();
  });

  test("debe de decrementar con el botón -1", () => {
    render(<App value={10} />);
    fireEvent.click(screen.getByText("-1"));
    fireEvent.click(screen.getByText("-1"));
    fireEvent.click(screen.getByText("-1"));
    expect(screen.getByText("7")).toBeTruthy();
  });

  test("debe de funcionar el botón de reset", () => {
    render(<App value={355} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    // fireEvent.click( screen.getByText('Reset') );
    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));

    expect(screen.getByText(355)).toBeTruthy();
  });
});
