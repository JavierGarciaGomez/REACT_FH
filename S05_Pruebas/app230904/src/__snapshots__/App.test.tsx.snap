// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <App /> should match the snapshot 1`] = `
<div>
  <h1>
    100
  </h1>
  <div>
    DIV
  </div>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
